src/impl_methods.rs

```diff
@@ -340,9 +340,9 @@ where
     ///
     /// **Panics** if an index is out of bounds or step size is zero.<br>
     /// (**Panics** if `D` is `IxDyn` and `info` does not match the number of array axes.)
-    pub fn slice<Do>(&self, info: &SliceInfo<D::SliceArg, Do>) -> ArrayView<'_, A, Do>
+    pub fn slice<I>(&self, info: &I) -> ArrayView<'_, A, I::OutDim>
     where
-        Do: Dimension,
+        I: CanSlice<D>,
```

Comment:
Potentially we'd be better off with this trait being named `SliceArg`, `SliceInfo` or something similar, direct and descriptive.

Reply:
Yeah, I don't really know what to call it. Updating the list from your earlier comment, we have the following:

- `Slice` (struct) 1D
- `AxisSliceInfo` (enum) 1D
- `SliceArg` (trait) 1D, used internally by `s![]` macro and hidden from the docs
- `SliceInfo` (struct) nD
- `CanSlice` (trait) nD
- `MultiSlice` (trait) nD
- `NewAxis` (struct) 1D

It would be nice for the following to have similar names:

- `CanSlice` and `MultiSlice`
- `AxisSliceInfo` and `SliceInfo`

`SliceArg`/`MultiSliceArg` would work. I've also considered `Slicer`/`MultiSlicer`. `SliceInfo`/`MultiSliceInfo` would be fine, but we'd need to rename the `SliceInfo` struct and probably `AxisSliceInfo`, too.

Do you have a preference?

Reply:
I think Slicer/MultiSlicer sound good then (the SliceArg pair would also works) because SliceInfo sounds good as a concrete struct.

Reply:
After thinking about it some more, `SliceArg`/`MultiSliceArg` sound a little more natural to me than `Slicer`/`MultiSlicer` in this context, so I've renamed `SliceArg` to `SliceNextDim`, `CanSlice` to `SliceArg`, and `MultiSlice` to `MultiSliceArg`.